src/OpenBinary/reader.jl

```diff
@@ -156,137 +147,157 @@ function PSRI.open(
+                    if i > first_relative_stage
+                        number_blocks[i-first_relative_stage] = offset - offsets[i-first_relative_stage]
+                    end
                 end
             end
+            
             block_total = maximum(number_blocks)
 
-            hour_discretization = floor(
-                Int,
-                block_total / if stage_type == PSRI.STAGE_WEEK
-                    168
-                elseif stage_type == PSRI.STAGE_MONTH
-                    if block_total % 672 == 0
-                        672
-                    elseif block_total % 720 == 0
-                        720
-                    else
-                        744
-                    end
-                elseif stage_type == PSRI.STAGE_DAY
-                    24
-                elseif stage_type == PSRI.STAGE_YEAR
-                    8760
-                else
-                    error("stage Type $stage_type not currently supported")
-                end,
-            )
+            hour_discretization = _get_hour_discretization(stage_type, block_total)
 
             if verbose_header
                 @show(block_total, length(number_blocks), number_blocks)
             end
         else
-            error("variable_by_hour = $variable_by_hour is invalid")
+            error("'variable_by_hour' must be either '0' or '1', not '$(variable_by_hour)'")
         end
     end
+
     agent_names = String[]
     agent_name_buffer = Vector{Cchar}(undef, name_length)
+
     for _ in 1:total_agents
-        skip_store(ioh, 4, skips, store_skips)
-        skip_store(ioh, 4, skips, store_skips)
+        skip(ioh, 4)
+        skip(ioh, 4)
+
         read!(ioh, agent_name_buffer)
         agent_name = strip(join(Char.(agent_name_buffer)))
         push!(agent_names, agent_name)
     end
+
     if verbose_header
         @show(agent_names)
     end
 
-    if single_binary
-        # nothing
-    else
+    if !single_binary
         close(ioh)
     end
 
-    BAD = "Bad header: "
+    if !(stage_total > 0)
+        error("'stage_total' must be a positive integer, not '$(stage_total)'")
+    end
 
-    @assert stage_total > 0
-    @assert first_relative_stage == 1 # todo improve this
-    @assert stage_total - first_relative_stage >= 0
+    # TODO: improve this
+    # if first_relative_stage != 1
+    #     @info   first_relative_stage
+    #     @assert false
+    # end
+
+    if !(first_relative_stage <= stage_total)
+        error("'first_relative_stage' must be less than or equal to 'stage_total'")
+    end
 
     if variable_by_series == 0
         # previous versions of the file simply leave the number of scenarios
         # as the number of scenarios in the study
         scenario_total = 1
     end
+
     if !(0 <= variable_by_series <= 1)
-        println(BAD * "variable_by_series = $variable_by_series, expected 0 or 1")
+        @warn("'variable_by_series' must be either 0 or 1, not $(variable_by_series)")
```

Comment:
Maybe this should be an error

Reply:
I just turned `print`s into `@warn`s in this commit. We can discuss that.